Compiling the extension locally

### Before submitting your bug report

- [*] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [*] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [*] I've seen the [troubleshooting guide](https://docs.continue.dev/troubleshooting) on the Continue Docs

### Relevant environment info

```
- OS: Ubuntu
- Node.js version: 20.11.0
- Continue version: release v1.0.5-vscode
```

### Description

I want to compile Continue extension locally, to obtain .vsix file from repository rather than using vscode. To do so, I try to follow the contributing guide ( https://github.com/continuedev/continue/blob/main/CONTRIBUTING.md ). 

However, packaging the extension via running ./scripts/install-dependencies.sh (first step in packaging) does not work, as continue cannot find module  '@continuedev/config-yaml' , which leads to errors. Adding     "@continuedev/config-yaml":"^1.0.3" to dependencies/package.json does not help.

In following next steps, when I go to the directory extensions/vscode and run " npm install" and than  "npm run package" (to package the extension), it throws error. 

The only place where error was mentioned was here https://github.com/continuedev/continue/issues/812 . Even though the fixes mentioned there were already included in this release, the problem persists.

After running  npm install @continuedev/config-yaml@1.0.78 (even though it was not included into the instruction, the version was chosen according to git repo) other problem arises:


### To reproduce

1. Clone https://github.com/continuedev/continue.git, release v1.0.5-vscode
2 ./scripts/install-dependencies.sh 
2.1  npm install @continuedev/config-yaml@1.0.78 ; ./scripts/install-dependencies.sh 
3. cd extensions/vscode @ npm install @ npm run package also does not work

### Log output

```Shell
For point 2:

(base) dimakarp1996@dimakarp1996-System-Product-Name:~/continue2/continue$ ./scripts/install-dependencies.sh
📦 Using nvm to set Node.js version to 20.11.0...
✅ Using Node.js v20.11.0 as required.
Installing root-level dependencies...

up to date, audited 243 packages in 690ms

98 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
Installing Core extension dependencies...
~/continue2/continue/core ~/continue2/continue

changed 1 package, and audited 1289 packages in 2s

187 packages are looking for funding
  run `npm fund` for details

10 vulnerabilities (1 low, 8 moderate, 1 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.

up to date, audited 3 packages in 868ms

found 0 vulnerabilities
~/continue2/continue
Installing GUI extension dependencies...
~/continue2/continue/gui ~/continue2/continue

removed 1 package, changed 1 package, and audited 961 packages in 3s

359 packages are looking for funding
  run `npm fund` for details

14 vulnerabilities (6 moderate, 5 high, 3 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

added 1 package, changed 1 package, and audited 962 packages in 2s

359 packages are looking for funding
  run `npm fund` for details

14 vulnerabilities (6 moderate, 5 high, 3 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

> build
> tsc && vite build

../core/config/ConfigHandler.ts:1:30 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ConfigResult } from "@continuedev/config-yaml";
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/ProfileLifecycleManager.ts:5:8 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

5 } from "@continuedev/config-yaml";
         ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/default.ts:1:28 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ConfigYaml } from "@continuedev/config-yaml";
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/load.ts:10:8 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

10 } from "@continuedev/config-yaml";
          ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/onboarding.ts:1:28 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ConfigYaml } from "@continuedev/config-yaml";
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/profile/ControlPlaneProfileLoader.ts:2:30 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

2 import { ConfigResult } from "@continuedev/config-yaml";
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/profile/IProfileLoader.ts:3:30 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

3 import { ConfigResult } from "@continuedev/config-yaml";
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/profile/LocalProfileLoader.ts:1:47 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ConfigResult, parseConfigYaml } from "@continuedev/config-yaml";
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/profile/PlatformProfileLoader.ts:1:49 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { AssistantUnrolled, ConfigResult } from "@continuedev/config-yaml";
                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/profile/PlatformProfileLoader.ts:132:41 - error TS7006: Parameter 'e' implicitly has an 'any' type.

132     if (this.configResult.errors?.find((e) => e.fatal)) {
                                            ~

../core/config/profile/doLoadConfig.ts:8:8 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

8 } from "@continuedev/config-yaml";
         ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/selectedModels.ts:1:27 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ModelRole } from "@continuedev/config-yaml";
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/validation.ts:1:39 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ConfigValidationError } from "@continuedev/config-yaml";
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/yaml/loadYaml.ts:15:8 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

15 } from "@continuedev/config-yaml";
          ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/yaml/loadYaml.ts:104:21 - error TS7006: Parameter 'error' implicitly has an 'any' type.

104   if (errors?.some((error) => error.fatal)) {
                        ~~~~~

../core/config/yaml/loadYaml.ts:182:43 - error TS7006: Parameter 'doc' implicitly has an 'any' type.

182   continueConfig.docs = config.docs?.map((doc) => ({
                                              ~~~

../core/config/yaml/loadYaml.ts:189:58 - error TS7006: Parameter 'mcpServer' implicitly has an 'any' type.

189     modelContextProtocolServers: config.mcpServers?.map((mcpServer) => ({
                                                             ~~~~~~~~~

../core/config/yaml/loadYaml.ts:226:28 - error TS7006: Parameter 'prompt' implicitly has an 'any' type.

226   config.prompts?.forEach((prompt) => {
                               ~~~~~~

../core/config/yaml/loadYaml.ts:346:34 - error TS7006: Parameter 'cp' implicitly has an 'any' type.

346     (config.context || []).find((cp) => cp.provider === "codebase")?.params ||
                                     ~~

../core/config/yaml/loadYaml.ts:358:12 - error TS7006: Parameter 'context' implicitly has an 'any' type.

358     ?.map((context) => {
               ~~~~~~~

../core/config/yaml/loadYaml.ts:372:14 - error TS7006: Parameter 'p' implicitly has an 'any' type.

372     .filter((p) => !!p) ?? []) as IContextProvider[];
                 ~

../core/config/yaml/loadYaml.ts:387:36 - error TS7006: Parameter 'server' implicitly has an 'any' type.

387     (config.mcpServers ?? []).map((server) => ({
                                       ~~~~~~

../core/config/yaml/models.ts:1:29 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ModelConfig } from "@continuedev/config-yaml";
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/yaml/models.ts:78:27 - error TS7006: Parameter 'c' implicitly has an 'any' type.

78   if (capabilities?.find((c) => c === "tool_use")) {
                             ~

../core/config/yaml/models.ts:85:27 - error TS7006: Parameter 'c' implicitly has an 'any' type.

85   if (capabilities?.find((c) => c === "image_input")) {
                             ~

../core/control-plane/client.ts:9:8 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

9 } from "@continuedev/config-yaml";
         ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/core.ts:42:28 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

42 import { ConfigYaml } from "@continuedev/config-yaml";
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/core.ts:395:17 - error TS7006: Parameter 'provider' implicitly has an 'any' type.

395         ?.find((provider) => provider.description.title === msg.data.title)
                    ~~~~~~~~

../core/core.ts:399:19 - error TS7006: Parameter 'url' implicitly has an 'any' type.

399           fetch: (url, init) =>
                      ~~~

../core/core.ts:399:24 - error TS7006: Parameter 'init' implicitly has an 'any' type.

399           fetch: (url, init) =>
                           ~~~~

../core/core.ts:664:10 - error TS7006: Parameter 't' implicitly has an 'any' type.

664         (t) => t.function.name === toolCall.function.name,
             ~

../core/core.ts:816:10 - error TS7006: Parameter 'model' implicitly has an 'any' type.

816         (model) => model.title === msg.data.title,
             ~~~~~

../core/core.ts:818:38 - error TS7006: Parameter 'model' implicitly has an 'any' type.

818       config.modelsByRole.chat.find((model) =>
                                         ~~~~~

../core/core.ts:896:10 - error TS7006: Parameter 'provider' implicitly has an 'any' type.

896         (provider) => provider.description.title === name,
             ~~~~~~~~

../core/core.ts:923:17 - error TS7006: Parameter 'url' implicitly has an 'any' type.

923         fetch: (url, init) =>
                    ~~~

../core/core.ts:923:22 - error TS7006: Parameter 'init' implicitly has an 'any' type.

923         fetch: (url, init) =>
                         ~~~~

../core/core.ts:935:25 - error TS7006: Parameter 'item' implicitly has an 'any' type.

935       return items.map((item) => ({
                            ~~~~

../core/data/log.ts:9:8 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

9 } from "@continuedev/config-yaml";
         ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/data/log.ts:108:32 - error TS7006: Parameter 'dataConfig' implicitly has an 'any' type.

108         config.data.map(async (dataConfig) => {
                                   ~~~~~~~~~~

../core/indexing/docs/DocsService.ts:1:30 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ConfigResult } from "@continuedev/config-yaml";
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/indexing/docs/DocsService.ts:1203:16 - error TS7006: Parameter 'doc' implicitly has an 'any' type.

1203               (doc) => (doc as any).startUrl !== siteIndexingConfig.startUrl,
                    ~~~

../core/indexing/docs/DocsService.ts:1265:16 - error TS7006: Parameter 'doc' implicitly has an 'any' type.

1265               (doc) => "startUrl" in doc && doc.startUrl !== startUrl,
                    ~~~

../core/llm/countTokens.ts:10:22 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

10 import { Rule } from "@continuedev/config-yaml";
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/llm/index.ts:1:33 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ModelRole, Rule } from "@continuedev/config-yaml";
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/llm/llms/stubs/ContinueProxy.ts:5:8 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

5 } from "@continuedev/config-yaml";
         ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/llm/rules/isRuleActive.ts:1:22 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { Rule } from "@continuedev/config-yaml";
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/llm/streamChat.ts:49:8 - error TS7006: Parameter 'sc' implicitly has an 'any' type.

49       (sc) => sc.name === command.name,
          ~~

../core/llm/streamChat.ts:69:24 - error TS7006: Parameter 'item' implicitly has an 'any' type.

69       addContextItem: (item) => {
                          ~~~~

../core/llm/streamChat.ts:77:15 - error TS7006: Parameter 'url' implicitly has an 'any' type.

77       fetch: (url, init) =>
                 ~~~

../core/llm/streamChat.ts:77:20 - error TS7006: Parameter 'init' implicitly has an 'any' type.

77       fetch: (url, init) =>
                      ~~~~

../core/protocol/core.ts:5:8 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

5 } from "@continuedev/config-yaml";
         ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/protocol/webview.ts:1:30 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ConfigResult } from "@continuedev/config-yaml";
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/util/GlobalContext.ts:3:27 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

3 import { ModelRole } from "@continuedev/config-yaml";
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/util/paths.ts:6:42 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

6 import { ConfigYaml, DevEventName } from "@continuedev/config-yaml";
                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/mainInput/Lump/EditBlockButton.tsx:1:28 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ConfigYaml } from "@continuedev/config-yaml";
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/mainInput/Lump/sections/ModelsSection.tsx:1:27 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ModelRole } from "@continuedev/config-yaml";
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/mainInput/Lump/sections/RulesSection.tsx:1:44 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { parseConfigYaml, type Rule } from "@continuedev/config-yaml";
                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/mainInput/Lump/sections/docs/DocsIndexingStatus.tsx:1:28 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ConfigYaml } from "@continuedev/config-yaml";
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/mainInput/Lump/sections/docs/DocsSection.tsx:1:33 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { parseConfigYaml } from "@continuedev/config-yaml";
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~

src/components/modelSelection/platform/shared.tsx:1:39 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ConfigValidationError } from "@continuedev/config-yaml";
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~

src/pages/config/MCPServersPreview.tsx:1:45 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ConfigYaml, parseConfigYaml } from "@continuedev/config-yaml";
                                              ~~~~~~~~~~~~~~~~~~~~~~~~~~

src/redux/slices/configSlice.ts:1:53 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ConfigResult, ConfigValidationError } from "@continuedev/config-yaml";
                                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~

src/redux/thunks/updateSelectedModelByRole.ts:1:27 - error TS2307: Cannot find module '@continuedev/config-yaml' or its corresponding type declarations.

1 import { ModelRole } from "@continuedev/config-yaml";
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 63 errors in 36 files.

For point 2.1

./core/config/yaml/loadYaml.ts:8:3 - error TS2305: Module '"@continuedev/config-yaml"' has no exported member 'isAssistantUnrolledNonNullable'.

8   isAssistantUnrolledNonNullable,
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../core/config/yaml/models.ts:69:34 - error TS2339: Property 'chatOptions' does not exist on type '{ provider: "continue-proxy"; apiKeyLocation: string; orgScopeId: string | null; onPremProxyUrl: string | null; name: string; model: string; apiKey?: string | undefined; apiBase?: string | undefined; ... 6 more ...; env?: Record<...> | undefined; } | { ...; }'.
  Property 'chatOptions' does not exist on type '{ provider: "continue-proxy"; apiKeyLocation: string; orgScopeId: string | null; onPremProxyUrl: string | null; name: string; model: string; apiKey?: string | undefined; apiBase?: string | undefined; ... 6 more ...; env?: Record<...> | undefined; }'.

69     baseChatSystemMessage: model.chatOptions?.baseSystemMessage,
                                    ~~~~~~~~~~~

                                    ~~~~~~~~~~~


For point 3:


npm run package

> continue@1.1.23 prepackage
> node scripts/prepackage.js

[info] Using target:  linux-x64
[info] Packaging extension for target  linux-x64
[info] npm install in extensions/vscode completed
[info] npm install in gui completed
[info] Copied gui build to JetBrains extension
Copied gui build to VSCode extension
/home/dimakarp1996/continue2/continue/extensions/vscode/scripts/prepackage.js:147
    throw new Error("gui build did not produce index.js");
          ^

Error: gui build did not produce index.js
    at /home/dimakarp1996/continue2/continue/extensions/vscode/scripts/prepackage.js:147:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

--- a/scripts/install-dependencies.sh
+++ b/scripts/install-dependencies.sh
@@ -61,13 +61,18 @@ setup_node_version
 echo "Installing root-level dependencies..."
 npm install
 
+echo "Building config-yaml..."
+pushd packages/config-yaml
+npm install
+npm run build
+popd
+
 echo "Installing Core extension dependencies..."
 pushd core
 ## This flag is set because we pull down Chromium at runtime
 export PUPPETEER_SKIP_DOWNLOAD='true'
 npm install
 npm link
-
 popd
 
 echo "Installing GUI extension dependencies..."
@@ -75,7 +80,6 @@ pushd gui
 npm install
 npm link @continuedev/core
 npm run build
-
 popd
 
 # VSCode Extension (will also package GUI)
@@ -86,18 +90,15 @@ npm install
 npm link @continuedev/core
 npm run prepackage
 npm run package
-
 popd
 
 echo "Installing binary dependencies..."
 pushd binary
 npm install
 npm run build
-
 popd
 
 echo "Installing docs dependencies..."
 pushd docs
 npm install
-
 popd

